modules/markup/markdown/math/block_parser.go

```diff
@@ -53,6 +53,12 @@ func (b *blockParser) Open(parent ast.Node, reader text.Reader, pc parser.Contex
 	}
 	idx := bytes.Index(line[pos+2:], endBytes)
 	if idx >= 0 {
+		// for case $$ ... $$ any other text
+		for i := pos + idx + 4; i < len(line); i++ {
+			if line[i] != ' ' && line[i] != '\n' {
+				return nil, parser.NoChildren
+			}
+		}
```

Comment:
What happens to `$$ a $$ xxxxxxxxx $$ b $$`?

Reply:
> What happens to `$$ a $$ xxxxxxxxx $$ b $$`?

In the current version, any text after a $$ block will be ignored because it is rendered as a block. So, I check the character after the second $$; if any text exists, it will not be rendered as a block.

In the current version, only `a` will be displayed ,  `xxxxxxx` and `b` will disappear.
After the fix,  `a`  , `xxxxxxx`, and `b` will all be displayed.
![圖片](https://github.com/go-gitea/gitea/assets/30816317/768c8485-785d-4f98-a1d0-682f320298d2)